### PR TITLE
Cb 4389 convert reverse proxy

### PR DESCRIPTION
**Pull request by DenisSinelnikov**

### CODE REVIEW

File: server/bundles/io.cloudbeaver.service.auth/src/io/cloudbeaver/service/auth/ReverseProxyConfigurator.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    private void migrateConfiguration(
+        @NotNull WebApplication application
+    ) {
+        if (!(application instanceof WebAuthApplication authApplication)) {
+            return;
+        }
+
+        SMAuthProviderCustomConfiguration smReverseProxyhProviderConfiguration =
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.service.auth/src/io/cloudbeaver/service/auth/ReverseProxyConfigurator.java]:
```suggestion
        SMAuthProviderCustomConfiguration smReverseProxyProviderConfiguration =
```

File: server/bundles/io.cloudbeaver.service.auth/src/io/cloudbeaver/service/auth/ReverseProxyConfigurator.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        if (!(application instanceof WebAuthApplication authApplication)) {
+            return true;
+        }
+
+        if (!authApplication.getAuthConfiguration().isAuthProviderEnabled(RPAuthProvider.AUTH_PROVIDER)) {
+            log.debug("Reverse proxy provider disabled, migration not needed");
+            return true;
+        }
+
+        if (authApplication.getAuthConfiguration().getAuthProviderConfiguration(RPAuthProvider.AUTH_PROVIDER) != null) {
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.service.auth/src/io/cloudbeaver/service/auth/ReverseProxyConfigurator.java]: check that any reverse proxy provider exists, not only by id

File: server/bundles/io.cloudbeaver.service.auth/src/io/cloudbeaver/service/auth/ReverseProxyConfigurator.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    private void migrateConfiguration(
+        @NotNull WebApplication application
+    ) {
+        if (!(application instanceof WebAuthApplication authApplication)) {
+            return;
+        }
+
+        SMAuthProviderCustomConfiguration smReverseProxyhProviderConfiguration =
+            authApplication.getAuthConfiguration().getAuthProviderConfiguration(RPAuthProvider.AUTH_PROVIDER);
+        boolean createNew = false;
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.service.auth/src/io/cloudbeaver/service/auth/ReverseProxyConfigurator.java]: redundant flag, all logic can be placed under `smReverseProxyhProviderConfiguration == null`

**Review by alexander-skoblikov — CHANGES_REQUESTED** [review]

File: server/bundles/io.cloudbeaver.service.auth/src/io/cloudbeaver/service/auth/ReverseProxyConfigurator.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
+package io.cloudbeaver.service.auth;
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.service.auth/src/io/cloudbeaver/service/auth/ReverseProxyConfigurator.java:17]: copyright

File: server/bundles/io.cloudbeaver.service.auth/src/io/cloudbeaver/service/auth/RPConstants.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package io.cloudbeaver.service.auth;
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.service.auth/src/io/cloudbeaver/service/auth/RPConstants.java:17]: copyright

File: webapp/packages/plugin-authentication/src/AuthenticationService.ts [hunk 6]
```diff
@@ -91,38 +90,31 @@ export class AuthenticationService extends Bootstrap {
...
-  private async logoutConfiguration(configuration: IUserAuthConfiguration): Promise<void> {
-    if (configuration.configuration.signOutLink) {
-      const id = `${configuration.configuration.id}-sign-out`;
+  // TODO handle all redirect links once we know what to do with multiple popups issue
+  private handleRedirectLinks(userLogoutInfo: UserLogoutInfo) {
+    const redirectLinks = userLogoutInfo.redirectLinks;
+
+    if (redirectLinks.length) {
+      const url = redirectLinks[0];
+      const id = `okta-logout-id-${uuid()}`;
```
Comment (devnaumov) [inline webapp/packages/plugin-authentication/src/AuthenticationService.ts]: Why should we mention okta here?

Comment (DenisSinelnikov) [reply webapp/packages/plugin-authentication/src/AuthenticationService.ts]: preparing branch before merge in devel, outdated

**Review by alexander-skoblikov — APPROVED** [review]

**Pull request merged**